insights/parsers/rsyslog_conf.py

```diff
@@ -77,3 +77,130 @@ def __len__(self):
+        self._module_details = {}
+        self._include_details = {}
+        self._input_details = {}
+        self._global_details = {}
+        self._log_details = {}
+
+        self.data = get_active_lines(content)
+        for line in self.data:
+            # Parse module and its configurations on two different line
+            if ('$' not in line) and (line.startswith('module')) and (not line.endswith(')')):
+                module_name = re.split(r'\W+', line)[2]
+                self._module_details[module_name] = {}
+            # Parse module and its configurations on two different line.
+            elif ('$' not in line) and ('(' not in line) and line.endswith(')'):
+                line_split = re.findall(r'[a-z.:/*A-Z0-9_]+', line)
+                log_level = line_split[0]
+                status = line_split[1]
+                self._module_details[module_name][log_level] = status
+            # Parse module and its configurations on one line.
+            elif ('$' not in line) and (line.startswith('module')) and '(' in line and (line.endswith(')')):
+                line_split = re.findall(r'[a-z:A-Z0-9_]+', line)
+                module_name = line_split[2]
+                self._module_details[module_name] = {}
+                if len(line_split) > 3:
+                    timestamp_format = line_split[3]
+                    timestamp = line_split[4]
+                    self._module_details[module_name][timestamp_format] = timestamp
+            # Parse included configration files and modes.
+            elif ('$' not in line) and (line.startswith('include')) and '(' in line and (line.endswith(')')):
+                line_split = re.findall(r'[a-z.:/*A-Z0-9_]+', line)
+                file_path = line_split[2]
+                self._include_details[file_path] = {}
+                file_mode = line_split[3]
+                modes = line_split[4]
+                self._include_details[file_path][file_mode] = modes
+            # Parse input module and its ports.
+            elif ('$' not in line) and (line.startswith('input')) and '(' in line and (line.endswith(')')):
+                line_split = re.findall(r'[a-z.:/*A-Z0-9_]+', line)
+                input_type = line_split[2]
+                port = line_split[4]
+                self._input_details[input_type] = port
+            # Parse global workspace.
+            elif ('$' not in line) and (line.startswith('global')) and '(' in line and (line.endswith(')')):
+                line_split = re.findall(r'[a-z.:/*A-Z0-9_]+', line)
+                global_dir = line_split[1]
+                path = line_split[2]
+                self._global_details[global_dir] = path
+            # Parse other configuations
+            else:
+                line_split = line.split()
+                self._log_details[line_split[0]] = line_split[1]
+
+    def __len__(self):
+        return len(self.data)
+
+    def __iter__(self):
+        for d in self.data:
+            yield d
```

Comment:
Do you need an iterator for the class?  It seems that you only use self.data to parse the content, and that a developer would not need the `__iter__` or `__len__` methods.